orchprovider/k8s/v1/util.go

```diff
@@ -193,6 +193,74 @@ func (p *VolumeMarkerBuilder) AddControllerStatuses(pod k8sApiV1.Pod) {
 	_ = p.AddMultiples(string(v1.JivaControllerStatusVK), status, true)
 }
 
+// AddControllerContainerStatuses is to fetch the current state of controller containers
+// inside the pod
+func (p *VolumeMarkerBuilder) AddControllerContainerStatuses(cp k8sApiV1.Pod) {
+	for _, current := range cp.Status.ContainerStatuses {
+		value := v1.NilVV
+		if current.State.Waiting != nil {
+			value = v1.ContainerWaitingVV
+		}
+
+		if current.State.Terminated != nil {
+			value = v1.ContainerTerminatedVV
+		}
+
+		if current.State.Running != nil {
+			if current.Ready == true {
+				value = v1.ContainerRunningVV
+			} else {
+				value = v1.ContainerNotRunningVV
+			}
+		}
+		_ = p.AddMultiples(string(v1.ControllerContainerStatusVK), string(value), true)
+	}
+}
+
+// AddReplicaContainerStatuses is to fetch the current state of replica containers
+// inside the pod
+func (p *VolumeMarkerBuilder) AddReplicaContainerStatuses(cp k8sApiV1.Pod) {
+	for _, current := range cp.Status.ContainerStatuses {
+		value := v1.NilVV
+		if current.State.Waiting != nil {
+			value = v1.ContainerWaitingVV
+		}
+
+		if current.State.Terminated != nil {
+			value = v1.ContainerTerminatedVV
+		}
+
+		if current.State.Running != nil {
+			if current.Ready == true {
+				value = v1.ContainerRunningVV
+			} else {
+				value = v1.ContainerNotRunningVV
+			}
+		}
+		_ = p.AddMultiples(string(v1.ReplicaContainerStatusVK), string(value), true)
+	}
+}
+
+// IsRunning to compare the state of all containers in a pod
+/*func (p *VolumeMarkerBuilder) IsRunning(keys []string) bool {
+
+	for i := 1; i < len(keys); i++ {
+		if keys[i] != keys[0] {
+			return false
+		}
+	}
+	return true
+}
+*/
+
+// IsRunning to compare the state of all containers in a pod
+func (p *VolumeMarkerBuilder) IsRunning(pv *v1.Volume) bool {
+	if pv.Annotations["openebs.io/controller-container-status"] == v1.ContainerRunningVV && pv.Annotations["openebs.io/replica-container-status"] == v1.ContainerRunningVV {
```

Comment:
Donot logic in this way.

Pass the set of KEYs. Loop the keys. Get the VolumeMarker. Get the corresponding Values []string or Value based on IsMultiple. 
Now check if any of the value does not have ContainerRunningVV.

Reply:
done